src/components/course/course-header/CourseImportantDates.jsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Col, Icon, Row, Stack,
+} from '@openedx/paragon';
+import { Calendar } from '@openedx/paragon/icons';
+import dayjs from 'dayjs';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { defineMessages, useIntl } from '@edx/frontend-platform/i18n';
+import { DATE_FORMAT } from '../data';
+import {
+  determineAllocatedCourseRunAssignmentsForCourse,
+  getSoonestEarliestPossibleExpirationData,
+  useCourseMetadata,
+  useRedeemablePolicies,
+} from '../../app/data';
+
+const messages = defineMessages({
+  importantDates: {
+    id: 'enterprise.course.about.page.important-dates.title',
+    defaultMessage: 'Important dates',
+    description: 'Title for the important dates section on the course about page',
+  },
+  enrollByDate: {
+    id: 'enterprise.course.about.page.important-dates.enroll-by-date',
+    defaultMessage: 'Enroll-by date',
+    description: 'Enroll-by date for the important dates section on the course about page',
+  },
+  courseStarts: {
+    id: 'enterprise.course.about.page.important-dates.course-starts',
+    defaultMessage: 'Course starts',
+    description: 'Course starts for the important dates section on the course about page in future tense',
+  },
+  courseStarted: {
+    id: 'enterprise.course.about.page.important-dates.course-started',
+    defaultMessage: 'Course started',
+    description: 'Course started the important dates section on the course about page in past tense',
+  },
+});
+
+const CourseImportantDates = () => {
+  const { courseKey } = useParams();
+  const { data: redeemableLearnerCreditPolicies } = useRedeemablePolicies();
+  const { data: courseMetadata } = useCourseMetadata();
+  const intl = useIntl();
+  const {
+    allocatedCourseRunAssignments,
+    allocatedCourseRunAssignmentKeys,
+    hasAssignedCourseRuns,
+  } = determineAllocatedCourseRunAssignmentsForCourse({
+    redeemableLearnerCreditPolicies,
+    courseKey,
+  });
+
+  const [searchParams] = useSearchParams();
+  const courseRunKey = searchParams.get('course_run_key')?.replaceAll(' ', '+');
+
+  if (!hasAssignedCourseRuns || (!!courseRunKey && !allocatedCourseRunAssignmentKeys.includes(courseRunKey))) {
+    return null;
+  }
+
+  const { date, soonestExpirationDateData } = getSoonestEarliestPossibleExpirationData({
+    assignmentObjectArray: allocatedCourseRunAssignments,
+    dateFormat: `${DATE_FORMAT} h:mm A`,
+  });
+  const soonestExpiringAllocatedAssignmentCourseStartDate = courseMetadata.availableCourseRuns.find(
+    (courseRun) => courseRun.key === soonestExpirationDateData?.contentKey,
+  )?.start;
+
+  const enrollByDate = date;
+  const courseStartDate = dayjs(soonestExpiringAllocatedAssignmentCourseStartDate).format(DATE_FORMAT);
+  const hasCourseStarted = dayjs(courseStartDate).isBefore(dayjs());
+
+  return (
+    <Stack gap={2} className="mt-4">
+      <h3>{intl.formatMessage(messages.importantDates)}</h3>
+      <section>
```

Comment:
If there are no dates (`enrollByDate` or `courseStartDate`), the `Stack`, `h3`, and `section` will all still render. Would it make sense to return early if neither of these are set so that we only render this stuff when there's at least one date to render?